### PR TITLE
Fixed broken tests, bad merge, regression in RC guid calculation (ProjectedRuntimeClass attrib), and made vector view GRCN more reliable

### PR DESCRIPTION
**Pull request by Scottj1s**

### CODE REVIEW

File: UnitTest/TestComponentCSharp_Tests.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ async Task InvokeWriteBufferAsync()
             await stream.WriteAsync(buffer);
         }
 
-        [Fact (Skip = "flakey")]
+        [Fact]
```
Comment (Scottj1s) [inline UnitTest/TestComponentCSharp_Tests.cs:63]: @j0shuams is confirming whether this test needed to be disabled (shouldn't have - has been passing for a while)

Comment (j0shuams) [reply UnitTest/TestComponentCSharp_Tests.cs:63]: Per builds, I think this was unnecessarily disabled. This change is good with me.

File: UnitTest/TestComponentCSharp_Tests.net5.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public TestCSharpNet5()
             TestObject = new Class();
         }
 
-        [Fact (Skip = "flakey")]
+        [Fact(Skip = "flakey")]
```
Comment (j0shuams) [inline UnitTest/TestComponentCSharp_Tests.net5.cs:21]: @Scottj1s did I misunderstand that these tests were passing now?

File: cswinrt/code_writers.h [hunk 3]
```diff
@@ -2313,7 +2313,20 @@ event % %;)",
                 set_simple_marshaler_type(m, type);
                 break;
             case category::interface_type:
-                m.marshaler_type = "MarshalInterface<" + m.param_type + ">";
+                if (settings.netstandard_compat) {
+                    if (get_mapped_type(type.TypeNamespace(), type.TypeName()) &&
+                        type.TypeNamespace() == "Windows.Foundation.Collections")
```
Comment (BenJKuhn) [inline cswinrt/code_writers.h:2318]: why does this not need to set m.marshaler_type, like the rest of the cases?

File: cswinrt/code_writers.h [hunk 4]
```diff
@@ -3517,7 +3530,21 @@ remove => %.Unsubscribe(value);
                     }
                     break;
                 case category::interface_type:
```
Comment (BenJKuhn) [inline cswinrt/code_writers.h:3533]: indenting issue? Inconsistent with previous case statement.

**Review by j0shuams — APPROVED** [review]

**Review by BenJKuhn — APPROVED** [review]

**Comment by Scottj1s** [comment]
thanks!!!

**Pull request merged**